octomap_server/src/OctomapServer.cpp

```diff
@@ -361,7 +361,15 @@ void OctomapServer::insertScan(const tf::Point& sensorOriginTf, const PCLPointCl
   {
     ROS_ERROR_STREAM("Could not generate Key for origin "<<sensorOrigin);
   }
+  // Generate octomap pointcloud
+  Pointcloud pc;
+  for (PCLPointCloud::const_iterator it = nonground.begin(); it != nonground.end(); ++it){
+    pc.push_back(it->x, it->y, it->z);
+  }
+  m_octree->insertPointCloud (pc, sensorOrigin, m_maxRange, true, true);
+  m_octree->updateInnerOccupancy();
 
+  /*
```

Comment:
Small change in this line, but you essentially replaced all of the specialized point cloud handling in the next 100 lines with the default OctoMap point cloud insertion, and hardcoded the option "discretize"=true. This could be done as speedup, but better as an optional parameter to octomap_server, because it creates different maps in the end (less information in them).

You also only insert the "nonground" point cloud, while the commented out 100 lines insert both ground and nonground, and give the option to filter out ground points while still using them for clearing obstacles - this is a significantly different behavior and different maps will be created in the end.